smslibrary/src/main/java/com/eis/smslibrary/SMSHandler.java

```diff
@@ -74,12 +74,26 @@ public static SMSHandler getInstance() {
     /**
      * Setup for the handler.
      *
-     * @param context current context.
+     * @param context current context. Must not be null
+     * @throws NullPointerException If the given context is null
      */
     public void setup(Context context) {
+        /*
+        null pointer exception is thrown because if the
+        context is null, this.context isn't, it's a WeakReference to a null object,
+        which later on causes problems
+         */
+        if(context == null) throw new NullPointerException("The given context must not be null!");
         this.context = new WeakReference<>(context);
     }
 
+    /**
+     * Clears the current context. Function used by tests
+     */
+    public void clearSetup(){
```

Comment:
Either you choose to make it protected or you remove the "Function used by tests" in the specifications, it's not good looking on a public method.